src/components/side_panel/table_panel/table_panel.ts

```diff
@@ -53,23 +53,33 @@ export class TablePanel extends Component<Props, SpreadsheetChildEnv> {
   static props = { onCloseSidePanel: Function, table: Object };
 
   state!: State;
+  /** We want to save the state of the "hasFilter" checkbox so that toggling the header off (which
+   * disable the filters), then toggling it back on doesn't change the "hasFilter" state*/
+  filtersEnabledIfPossible: boolean = false;
 
   setup() {
     const sheetId = this.env.model.getters.getActiveSheetId();
     this.state = useState({
       tableZoneErrors: [],
       tableXc: this.env.model.getters.getRangeString(this.props.table.range, sheetId),
     });
+    this.filtersEnabledIfPossible = this.props.table.config.hasFilters;
 
     onWillUpdateProps((nextProps: Props) => {
       if (this.props.table.id !== nextProps.table.id) {
         const sheetId = this.env.model.getters.getActiveSheetId();
         this.state.tableXc = this.env.model.getters.getRangeString(nextProps.table.range, sheetId);
         this.state.tableZoneErrors = [];
+        this.filtersEnabledIfPossible = nextProps.table.config.hasFilters;
```

Comment:
for later: I'm thinking about adding a way for a side panel to define a `t-key` (with the registry). It would allow to remove a bunch of `onWillUpdateProps`